pkg/sql/privilege/privilege.go

```diff
@@ -42,6 +42,7 @@ const (
 	USAGE      Kind = 9
 	ZONECONFIG Kind = 10
 	CONNECT    Kind = 11
+	RULE       Kind = 12
```

Comment:
@otan do you have thoughts on whether it would be better to not add RULE here? Given that we don't support rules at all, maybe it would be misleading to have this privilege?

Reply:
but we don't want to actually allow anyone to accidentally GRANT RULE anywhere -- would that be possible if it were added here?

Reply:
I think validate would fail -- https://github.com/cockroachdb/cockroach/blob/904f8e129d87fdcefe84ac27ca5d7a35d6b40324/pkg/sql/privilege/privilege.go#L242

We should add a test for that though - @sunisdown can you validate that we can't grant RULE privilege to a table?

Reply:
@otan  do we have similar tests? I can add a test for that, but I don't know how to make it.

Reply:
take a look at https://github.com/cockroachdb/cockroach/blob/cf41937bf1bf3df1f8fb398dace10a7a2383abd7/pkg/sql/logictest/testdata/logic_test/grant_table#L1607

Reply:
i just tried this and it works on PG13 so maybe we should allow it?
i'm ok either way, @rafiss wdyt?

Reply:
very strange! hm ok, we may as well allow it, since the motivation for this issue was initially all about compatbility anyway (a user in the field has a tool that was looking up RULE this way).

Reply:
in that case, we need to add this to this line here:

https://github.com/cockroachdb/cockroach/blob/904f8e129d87fdcefe84ac27ca5d7a35d6b40324/pkg/sql/privilege/privilege.go#L69

and add a test at the same place that it can be granted and inspected.

Reply:
actually, no, that's wrong, in postgresql that command works but is a no-op (the privilege never gets added):

```
otan=# grant rule on table tbl to otan;
GRANT

otan=# SELECT grantee, privilege_type
FROM information_schema.role_table_grants
WHERE table_name='tbl';
 grantee | privilege_type
---------+----------------
 otan    | INSERT
 otan    | SELECT
 otan    | UPDATE
 otan    | DELETE
 otan    | TRUNCATE
 otan    | REFERENCES
 otan    | TRIGGER
(7 rows)

otan=# select has_table_privilege('tbl'::regclass, 'RULE');
 has_table_privilege
---------------------
 f
(1 row)
```

so we should do the same.